books.php

```diff
@@ -15,6 +15,85 @@ public static function getInstance()
         }
         return self::$instance;
     }
+    public function auth($login, $password)
+    {
+        $stmt = $this->link->prepare("SELECT id FROM users WHERE BINARY login = :login AND BINARY password = :password");
+        $stmt->execute([':login' => $login, ':password' => $password]);
+        if ($stmt->rowCount() > 0) {
+            return true;
+        }
+        return false;
+    }
+    public function add($data)
+    {
+        $i = 0;
+        $arrayTable = ['name' => 0, 'genre' => 0, 'pages' => 0];
+        if ((count($data, COUNT_RECURSIVE) - count($data)) > 0) {

```

Comment:
You can change this condition check to array_intersect(), check final array length and then 
combine final array foreach operation with bindParam and form sql query.